view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/ViewGeneratorLauncher.java

```diff
@@ -91,4 +97,19 @@ public String getJobConfigKey() {
   private Config getJobConfig(Map<String, Object> properties) {
     return (Config) properties.get(getJobConfigKey());
   }
+
+  private AvroFieldValuePartitioner<GenericRecord> getPartitioner(
+      Map<String, Object> properties, String topic) {
+    AvroFieldValuePartitioner<GenericRecord> partitioner = null;
+    String topicsStr = (String) properties.get("avro.field.value.partitioner.enabled.topics");
+    if (topicsStr != null) {
+      boolean enabledForTopic =
+          Splitter.on(",").trimResults().splitToStream(topicsStr).anyMatch(topic::equals);
+      if (enabledForTopic) {
+        partitioner = new AvroFieldValuePartitioner<>();
+        partitioner.configure(properties);
```

Comment:
If we're going to own constructing and configuring it, we should combine into the constructor.

Alternatively, we could make this more generic and just take a partitioner class name per topic (so we look up `<topicName>.partitioner.class`, and if unset, return null, else construct.

Reply:
constructor - fixed it.

config - discussed